File: php/src/Types/PageInfo.php
<?php

declare(strict_types=1);

namespace PdfOxide\Types;

/**
 * Information about a PDF page.
 */
readonly class PageInfo
{
    public function __construct(
        public int $pageIndex,
        public float $width,
        public float $height,
        public int $rotation = 0,
        public int $fontCount = 0,
        public int $imageCount = 0,
        public int $annotationCount = 0
    ) {}

    /**
     * Get page dimensions as Rect.
     */
    public function getDimensions(): Rect
    {
        return new Rect(0, 0, $this->width, $this->height);
    }

    /**
     * Get page size in millimeters.
     */
    public function getSizeMm(): array
    {
        return [
            'width' => round($this->width * 0.352778, 1),
            'height' => round($this->height * 0.352778, 1),
        ];
    }

    /**
     * Get aspect ratio.
     */
    public function getAspectRatio(): float
    {
        return $this->height > 0 ? $this->width / $this->height : 0;
    }

    /**
     * Get total content count.
     */
    public function getTotalContentCount(): int
    {
        return $this->fontCount + $this->imageCount + $this->annotationCount;
    }

    /**
     * Convert to array format.
     */
    public function toArray(): array
    {
        return [
            'page_index' => $this->pageIndex,
            'width' => $this->width,
            'height' => $this->height,
            'rotation' => $this->rotation,
            'font_count' => $this->fontCount,
            'image_count' => $this->imageCount,
            'annotation_count' => $this->annotationCount,
            'size_mm' => $this->getSizeMm(),
            'aspect_ratio' => $this->getAspectRatio(),
        ];
    }
}

File: ruby/lib/pdf_oxide/types/page_info.rb
# frozen_string_literal: true

module PdfOxide
  module Types
    # Information about a PDF page
    class PageInfo
      attr_reader :index, :width, :height, :rotation, :page_count

      # Initialize page info
      # @param index [Integer] Page index
      # @param width [Float] Page width
      # @param height [Float] Page height
      # @param rotation [Integer] Page rotation (0, 90, 180, 270)
      # @param page_count [Integer] Total page count
      def initialize(index:, width:, height:, rotation: 0, page_count: 0)
        @index = index
        @width = width
        @height = height
        @rotation = rotation
        @page_count = page_count
      end

      # Get aspect ratio
      # @return [Float] Width/Height ratio
      def aspect_ratio
        @height.zero? ? 0.0 : (@width.to_f / @height)
      end

      # Convert to hash
      # @return [Hash] Page info as hash
      def to_h
        {
          index: @index,
          width: @width,
          height: @height,
          rotation: @rotation,
          page_count: @page_count,
          aspect_ratio: aspect_ratio
        }
      end

      # String representation
      # @return [String] Page info as string
      def to_s
        "PageInfo(index=#{@index}, width=#{@width}, height=#{@height}, rotation=#{@rotation})"
      end

      # Inspect representation
      # @return [String] Page info inspection
      def inspect
        to_s
      end
    end
  end
end
